### ISSUE TITLE
Upgrading from 3.1.11 to 3.1.31

### ISSUE THREAD

**Issue by zerocarbthirty** [smarty-php/smarty#419]
Hello, I am upgrading an application which uses Smarty from 3.1.11 to 3.1.31 and I am having an odd problem. I have a class for my application which uses Smarty.

```
<?php
require_once('smarty_new/Smarty.class.php');
class lsmarty extends Smarty {
    function __construct( ) {
        parent::__construct();
        $lang = "en_us";
	$this->left_delimiter = '{{';
	$this->right_delimiter = '}}';
	$this->force_compile = true;
	$this->setTemplateDir('tpls/'.$lang);
	$this->setConfigDir('configs/');
	$this->setCompileDir('cache/smarty_templates_c/');
	$this->setCacheDir('cache/smarty_cache/');
	$this->caching = Smarty::CACHING_OFF;
```

Here is one of the php scripts:

```
<?php
require_once('lsmarty.class.php');
$smarty = new lsmarty( );
$smarty->assign('reason', $reason);
$smarty->display('extends:login.tpl');
```

When requring the new version of Smarty the output is simply:

{extends file='extends:login.tpl' extends_resource=true}

When requiring the original version the output is the page as expected.

Any ideas?

**Comment by uwetews** [comment]
The fix is now in the master branch and will later be included in 3.1.32

However using the extends: resource with a sigle template file is not needed and does create overhead.

Just use
```php
$smarty->display('login.tpl');
```

**Comment by zerocarbthirty** [comment]
Migrated to Twig. Thanks for reply though.
